auth/ipa_2fa.adoc

```diff
@@ -0,0 +1,86 @@
+3.  <<configuring-external-auth, Configuring External Authentication>>
+4.  <<enabling-2fa, Enabling 2-Factor Authentication>>
+5.  <<using-2fa, Using 2-Factor Authentication>>
+
+[[requirements]]
+== Requirements
+
+The following is needed in order to enable 2-Factor-Authentication to the Appliance:
+
+* A CentOS/RHEL 7.1 based Appliance
+* IPA Server based on FreeIPA 4.1.0 or later
+
+[[configure-admin-ui]]
+== Configure Administrative UI 
+
+Login as admin, then in _Configure->Configuration->Authentication_
+
+* Set mode to External (httpd)
+* Check: _Get User Groups from External Authentication (httpd)_
+* Do Not Check: _Enable Single Signon_
+** *Note*: Kerberos SSO with OTP is not supported in the current release.
+* Click Save.
+
+The above steps need to be done on each UI and WebService enabled appliance.
+
+in _Configure->Configuration->Access Control_
+
+* Make sure the user's groups are created on the Appliance and appropriate roles assigned to those groups.
+
+[[configure-external-auth]]
+== Configure External Authentication
+
+Using the Appliance Console, simply enable external authentication as you currently 
+would to IPA.
+
+1. Login to console as _root_
+2. Run _appliance_console_
+3. Summary screen should show *External Auth* as _not configured_, Press any key
+4. From the Advanced Setting menu, select the menu item *Configure External Authentication (httpd)*
+5. Enter the FQDN of the IPA Server, i.e. _ipaserver.test.company.com_
+6. Enter the IPA Server domain, i.e. _test.company.com_
+7. Enter the IPA Server realm, i.e. _TEST.COMPANY.COM_
+8. Press enter to select the default IPA Server Principal, i.e. _admin_
+9. Enter the Password of the IPA Server Principal
+10. Review details, and Enter *y* to proceed.
+
+[[enabling-2fa]]
+== Enabling 2-Factor Authentication
+
+Enabling 2-Factor Authentication is done using the IPA administrative UI. 
+
+* The IPA administrator needs to change the User authentication types for the user 
+from Password or Radius to _Two factor authentication (password + OTP)_
+* The previous can be done either per user or for all users from the IPA Server
+global Configuration under User Options.
+* The user logs into IPA, then from the OTP Tokens tab,
+add an OTP token - Time-based (TOTP) or Counter-based (HOTP).
+* Specifying a Description for the OTP, then clicking add, brings up a QR Code
+which can then be scanned into an app like *FreeOTP* for example.
+* Then from the main IPA login screen, the user can Synchronize the OTP Token by
+specifying their Username, password and then two OTP's generated from the *FreeOTP* app.
+* After which, the user can login to FreeIPA using their Password+OTP.
+
+[[using-2fa]]
+== Using 2-Factor Authentication
+
+Once the above is done, the user now needs to specify their IPA Username and
+their Password followed by their One-Time-Password as generated by the *FreeOTP* app,
+when:
+
+* Logging into the Appliance Administrative UI
+* Accessing the REST API
+
```

Comment:
@abellotti After you answer the question above, I will merge this.  Then, I will go in and do the formatting and grammatical edits.